pandas/tests/indexing/test_loc.py

```diff
@@ -968,3 +968,11 @@ def test_loc_getitem_label_list_integer_labels(
     expected = df.iloc[:, expected_columns]
     result = df.loc[["A", "B", "C"], column_key]
     tm.assert_frame_equal(result, expected, check_column_type=check_column_type)
+
+
+def test_loc_upcast_int():
+    # GH 6485
+    result = DataFrame({"a": range(10)}, dtype="i4")
+    result.loc[10] = 99
+    expected = DataFrame({"a": list(range(10)) + [99]})
```

Comment:
can you specify the expected dtype here?  im actually not clear on why this is "fixed".  casting to int64 is better than casting to float64 like it did in the OP in #6485, but dont we want the dtype to stay int32?

Also, there is mention in that issue of an empty case that is still broken.  should that issue stay open or new issue for the still-broken case?

Reply:
Good point. Ideally we should maintain int32 here. I guess that issue is not totally solved and it leave it open for both cases